File: src/pb_100.cc
#include <iostream>
#include <math.h>
#include <stdio.h>

using namespace std;

int main() {
	long double i = 1000000000000L;
	long double r;
	while ((r=(1/2+sqrt(i*(i-1)/2+1/4))!=floor(r)) {
		++i;
	}
	printf("i = %Lf and number of blue discs = %Lf\n", i, r);
	return 0;
}

File: src/pb100.py
from math import sqrt

def isSquare(n):
	r = int(sqrt(n))
	return r**2==n

p=21
n=120

while True:
	d = 2*n**2 - 2*n +1
	while not isSquare(d):
		n+=1
		d=2*n**2-2*n+1
	if n>1e12:
		break
	p, n = n, int(n**2/p)

print((1+sqrt(d))/2)

